overlord/snapstate/backend/snapdata.go

```diff
@@ -108,8 +109,14 @@ func copySnapData(oldSnap, newSnap *snap.Info) (err error) {
 	return nil
 }
 
-// Lowlevel copy the snap data (but never override existing data)
+// Lowlevel copy the snap data
 func copySnapDataDirectory(oldPath, newPath string) (err error) {
+	if osutil.FileExists(newPath) {
+		if err := os.RemoveAll(newPath); err != nil {
+			return err
+		}
+	}
+
 	if _, err := os.Stat(oldPath); err == nil {
```

Comment:
This should probably be moved upfront, before we test for the existence of newPath. Seems bad to act so aggressively, removing the data, when we cannot actually copy anything else.

Might also make use of osutil helpers.